tests/ValueParsers/DecimalParserTest.php

```diff
@@ -123,4 +123,43 @@ public function testUnlocalization() {
 		$this->assertEquals( '20000000', $value->getValue() );
 	}
 
+	public function provideSplitDecimalExponent() {
+		return array(
+			'no exponent' => array( '1.2', '1.2', 0 ),
+			'exponent' => array( '1.2E3', '1.2', 3 ),
+			'negative exponent' => array( '+1.2e-2', '+1.2', -2 ),
+			'positive exponent' => array( '-12e+3', '-12', 3 ),
```

Comment:
I can think of a lot more test cases:
* `'leading zero' => array( '12e+09', '12', 9 )`
* `'trailing decimal point' => array( '12.e+3', '12.', 3 )`
* `'leading decimal point' => array( '.12e+3', '.12', 3 )`
* `'space' => array( '12 e+3', '12 ', 3 )`
* `'x10 syntax' => array( '12x10^3', '12', 3 )`
* `'comma' => array( '12e3,4', '12', 34 )`